File: rxhttp/src/main/java/rxhttp/wrapper/parse/TypeParser.java
package rxhttp.wrapper.parse;

import java.lang.reflect.Type;

import rxhttp.wrapper.utils.TypeUtil;


/**
 * User: ljx
 * Date: 2022/10/19
 * Time: 20:17
 */
public abstract class TypeParser<T> implements Parser<T> {

    protected final Type[] types;

    public TypeParser() {
        types = TypeUtil.getActualTypeParameters(getClass());
    }

    protected TypeParser(Type... types) {
        this.types = types;
    }
}

File: rxhttp/src/main/java/rxhttp/wrapper/parse/TypeParser.kt
package rxhttp.wrapper.parse

import rxhttp.wrapper.utils.getActualTypeParameters
import java.lang.reflect.Type

/**
 * This class stores multiple types in an array of [types]
 * User: ljx
 * Date: 2021/05/23
 * Time: 19:32
 */
abstract class TypeParser<T> : Parser<T> {

    @JvmField
    protected var types: Array<out Type>

    constructor() {
        types = getActualTypeParameters(javaClass)
    }

    constructor(vararg types: Type) {
        this.types = types
    }
}